### DISCUSSION TITLE
Should we use the Azure CLI for things that aren't normally supported via PS commands?

### DISCUSSION THREAD

**Discussion by stijnmoreels** [arcus-azure/arcus.scripting#213] [Ideas]
For some things, we use the Azure REST API to get or change behavior that isn't directly supported as PowerShell commands. Should we consider using the Azure CLI for this? As some of these things are supported there (for example: the setting of variables in variable groups).

**Comment by fgheysels** [comment]
My first thought would be: yes.   The Azure CLI is really straightforward and easy to use.  (I like it more then the Azure Powershell commands :) ).

But, I think there's a caveat:  when creating a Powershell function that uses Azure CLI commands internally, and you want to use your Powershell command in an Azure DevOps pipeline, you'll need to make sure that you're calling the Powershell command in an AzureCLI task.   (Offcourse, this is no big deal, but I think it needs to be documented so that ppl are aware of it when they want to use that Powershell command).

**Reply by fgheysels** [reply]
Any ideas on this ?   
Should we document that a certain script uses the Azure CLI internally, and that it must be used inside an AzureCLI task in a pipeline ?  Any other ideas ?

@pim-simons  @stijnmoreels

**Reply by pim-simons** [reply]
Sounds like a good idea to me, as long as we make sure it is clear in the documentation 👍🏻

**Reply by stijnmoreels** [reply]
Maybe there's a way to circumvent the problem of using a AzureCLI task, bc that would be rather troublesome I think for consumers to use. Especially if the only(?) added-value is to have more Azure support in the commands. It adds another dependency to the scripting, and I'm not entirely convinced if it's worth the while. Maybe we will come across some other problems and missing links in our codebase and we can have more reasons to include AzureCLI in our library. IDK.

**Comment by fgheysels** [comment]
I personally ran into an issue a while ago where I was mixing up Azure CLI and Azure Powershell tasks inside one `AzurePowershell` task in DevOps and came into trouble because the CLI commands couldn't be executed anymore because of an expired token.  (This was on self-hosted agent).

I personally would say that we should not use Azure CLI commands in our Arcus scripts.
And if we do, I think we should migrate everything so that Azure CLI is used instead of Azure Powershell commands.  In that case, we must make sure that we specify in the documentation that the Arcus scripts must be run inside `AzureCLI` tasks, and no longer inside `AzurePowershell` commands.

**Reply by stijnmoreels** [reply]
Ok, so maybe it's better to leave it as-is for now, then.
